Docker/Dockerfile

```diff
@@ -0,0 +1,47 @@
+# starting from a base image of Ubuntu
+FROM ubuntu:latest
+
+# **** Base System Initialization and Setup  ****
+
+# Install the required libraries:
+# - Gcc for compiler
+# - Git for source code access
+# - Cmake for build configuration
+# - Pkg_config to aide CMake in package sourcing see: https://autotools.io/pkgconfig/pkg_check_modules.html
+# - libssl for access to build requirements for encryption protocols
+RUN apt-get update \
+	&& apt-get upgrade -y \
+	&& apt-get install -y gcc git cmake pkg-config libssl-dev 
```

Comment:
Awesome feedback! Will make the requested changes and resubmit. 

A quick note: a Docker base image is (intentionally) light, but I think going with `build_essentials` rather than gcc makes *alot* of sense. I would suggest we keep a collection of Dockerfiles in a separate directory (to allow for building against multiple base images: e.g. Alpine Linux which is small but uses a modified gLibc). However I think using the base Dockerfile in the main directory is very good idea - as life become `git pull && docker built ...`